packages/scheduled-tasks/src/lib/strategies/ScheduledTaskSQSStrategy.ts

```diff
@@ -0,0 +1,88 @@
+import { container } from '@sapphire/framework';
+import type { SQS } from 'aws-sdk';
+import { randomBytes } from 'crypto';
+import { Consumer, ConsumerOptions } from 'sqs-consumer';
+import { Producer } from 'sqs-producer';
+import type { ScheduledTaskCreateRepeatedTask, ScheduledTasksTaskOptions } from '../types';
+import type { ScheduledTaskBaseStrategy } from '../types/ScheduledTaskBaseStrategy';
+
+export interface ScheduledTaskSQSStrategyMessageBody {
+	task: string;
+	payload?: unknown;
+	options: ScheduledTasksTaskOptions;
+}
+
+export class ScheduledTaskSQSStrategy implements ScheduledTaskBaseStrategy {
+	public readonly options: ConsumerOptions;
+
+	private producer: Producer;
+
+	public constructor(options: ConsumerOptions) {
+		this.options = options;
+		this.producer = Producer.create(this.options);
+	}
+
+	public connect() {
+		const consumer = Consumer.create({
+			...this.options,
+			// handleMessage requires a promise method as value. But we don't need to..
+			// eslint-disable-next-line @typescript-eslint/require-await
+			handleMessage: async (message) => void this.handleMessage(message),
+			// eslint-disable-next-line @typescript-eslint/require-await
+			handleMessageBatch: async (messages) => void this.handleBatch(messages)
+		});
+		consumer.start();
```

Comment:
Does this return something, by the way? Or is it just void?

Also, small style nit:

```suggestion

		consumer.start();
```

Reply:
![image](https://user-images.githubusercontent.com/4019718/147417159-5c8ec684-cda4-4ff1-b90a-4521e31296ff.png)